demonstrations/tutorial_rosalin.py

```diff
@@ -0,0 +1,598 @@
+algorithms, by randomly sampling operators from the cost Hamiltonian.
+
+Background
+----------
+
+While a large number of results in variational quantum algorithms focus on the
+choice of circuit ansatz, cost function, gradient computation, or initialization strategy,
+the optimization strategy --- an important component affecting both convergence time and
+quantum resource dependence --- is not as frequently considered. Instead, common
+'out-of-the-box' classical optimization techniques, such as gradient-free
+(COBLYA, Nelder-Mead), gradient-descent, and Hessian-free (L-BFGS) tend to be used.
+
+However, for variational algorithms such as :doc:`VQE </demos/tutorial_vqe>`, which involve evaluating
+a large number of non-commuting operators in the cost function, decreasing the number of
+quantum evaluations required for convergence, while still minimizing shot noise, can
+be delicate balance.
+
+Recent work has highlighted that 'quantum-aware' optimization techniques
+can lead to marked improvements when training variational quantum algorithms:
+
+* :doc:`/demos/tutorial_quantum_natural_gradient` descent by Stokes et al. [#stokes2019]_, which
+  takes into account the quantum geometry during the gradient-descent update step.
+
+* The work of Ryan Sweke et al. [#sweke2019]_, which shows
+  that quantum gradient descent with a finite number of shots is equivalent to
+  `stochastic gradient descent <https://en.wikipedia.org/wiki/Stochastic_gradient_descent>`_,
+  and has guaranteed convergence. Furthermore, combining a finite number of shots with
+  weighted sampling of the cost function terms leads to :doc:`/demos/tutorial_doubly_stochastic`.
+
+* The iCANS (individual Coupled Adaptive Number of Shots) optimization technique by
+  Jonas Kuebler et al. [#kubler2020]_ adapts the number
+  of shots measurements during training, by maximizing the expected gain per shot.
+
+In this latest result by Arrasmith et al. [#arrasmith2020]_, the
+idea of doubly stochastic gradient descent has been used to extend the iCANS optimizer,
+resulting in faster convergence. Let's explore their results.
+
+Weighted random sampling
+------------------------
+
+Consider a Hamiltonian :math:`H` expanded as a weighted sum of operators :math:`h_i` that can
+be directly measured:
+
+.. math:: H = \sum_{i=1}^N c_i h_i.
+
+Due to the linearity of expectation values, the expectation value of this Hamiltonian
+can be expressed as the weighted sum of each individual term:
+
+.. math:: \langle H\rangle = \sum_{i=1}^N c_i \langle h_i\rangle.
+
+In :doc:`doubly stochastic gradient descent demonstration </demos/tutorial_doubly_stochastic>`,
+we estimated this expectation value by **uniformly sampling** a subset of the terms
+at each optimization step, and evaluating each term by using the same finite number of shots
+:math:`N`.
+
+However, what happens if we use a weighted random approach to determine how to distribute

```

Comment:
"weighted random" sounds a bit unusual
```suggestion
However, what happens if we use a weighted approach to determine how to distribute

```